### PR TITLE
Combine PGs in buffermgrd

### PR DESCRIPTION
**Pull request by bingwang-ms**
Signed-off-by: bingwang <wang.bing@microsoft.com>

**What I did**
This PR is to combine PGs in `BUFFER_PG_TABLE`.

**Why I did it**
PR https://github.com/Azure/sonic-swss/pull/2171 converted the hardcoded lossless PG `3-4` into dynamic pgs, which caused compatibility issues.
This PR is to make it back compatible by combining `PG 3` and `PG 4` into `3-4` .
Also, PGs `2,3,4,6` is combined to `PG:2-4` and `PG:6`.

For example, before this PR, the `BUFFER_PG_TABLE` is as below
```
root@530a986b4d00:/# redis-cli -n 0 keys 'BUFFER_PG_TABLE:Ethernet0*' 
1) "BUFFER_PG_TABLE:Ethernet0:2"
2) "BUFFER_PG_TABLE:Ethernet0:6"
3) "BUFFER_PG_TABLE:Ethernet0:0"
4) "BUFFER_PG_TABLE:Ethernet0:3"
5) "BUFFER_PG_TABLE:Ethernet0:4"
```
After this PR
```
root@530a986b4d00:/# redis-cli -n 0 keys 'BUFFER_PG_TABLE:Ethernet0*' 
1) "BUFFER_PG_TABLE:Ethernet0:2-4"
2) "BUFFER_PG_TABLE:Ethernet0:6"
3) "BUFFER_PG_TABLE:Ethernet0:0"

```

**How I verified it**
Verified by vs test.

**Details if related**

### CODE REVIEW

**Review by stephenxs — DISMISSED** [review]
LGTM.

**Comment by neethajohn** [comment]
LGTM. Let me discuss with @prsunny as well before we merge this

**Comment by neethajohn** [comment]
LGTM. Let me discuss with @prsunny as well before we merge this

**Review by neethajohn — APPROVED** [review]

File: cfgmgr/buffermgr.cpp [hunk 1]
```diff
@@ -114,6 +115,65 @@ string BufferMgr::getPgPoolMode()
     return "";
 }
 
+vector<string> BufferMgr::combinePGs(const vector<string> &pgs)
+{
```
Comment (stephenxs) [inline cfgmgr/buffermgr.cpp:119]: We have an out-of-box method doing the same thing [set<string> Orch::generateIdListFromMap(unsigned long idsMap, sai_uint32_t maxId)](https://github.com/Azure/sonic-swss/blob/master/orchagent/orch.cpp#L735)
You may want to leverage it.

Comment (bingwang-ms) [reply cfgmgr/buffermgr.cpp]: Great. I update code to use the existing one. Thanks

File: cfgmgr/buffermgr.cpp [hunk 2]
```diff
@@ -251,7 +269,7 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port)
         SWSS_LOG_NOTICE("Reusing existing profile '%s'", buffer_profile_key.c_str());
     }
 
-    for (auto lossless_pg : lossless_pgs)
+    for (auto lossless_pg : lossless_pg_combinations)
```
[the remark this answers is not in the document]
Comment (bingwang-ms) [reply cfgmgr/buffermgr.cpp:272]: Good catch. Thanks

File: cfgmgr/buffermgr.cpp [hunk 3]
```diff
@@ -264,7 +282,7 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port)
             if ((fvField(prop) == "profile") && (profile_ref == fvValue(prop)))
             {
                 SWSS_LOG_NOTICE("PG to Buffer Profile Mapping %s already present", buffer_pg_key.c_str());
-                continue;
+                return task_process_status::task_success;
```
Comment (neethajohn) [inline cfgmgr/buffermgr.cpp:285]: we shouldn't return, we still need to check all lossless pgs. You can set a flag here and then outside this L280 loop, skip applying the pg if the flag is set

Comment (bingwang-ms) [reply cfgmgr/buffermgr.cpp]: Oh, I'm aware the low-level mistake just now. Please help check again. Thanks for pointing it out.

**Review by neethajohn — DISMISSED** [review]

**Review by neethajohn — APPROVED** [review]

**Review by lolyu — APPROVED** [review]

**Review by richardyu-ms — APPROVED** [review]

**Pull request merged**